### PR TITLE
Add test and implementation to fix potential NPE in pre-show resources

### PR DESCRIPTION
**Pull request by tadgh**
* Prevent hook invocation if the resolved resources would be null. 
* Prevent NPE in downstream interceptors. 
* Add test and changelog

### CODE REVIEW

**Comment by patrick-vachon-smilecdr** [comment]
praise: That looks like the easiest way to fix that! 

Review done.

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/binary/interceptor/BinaryStorageInterceptor.java [hunk 1]
```diff
@@ -256,30 +256,37 @@ public void preShow(IPreResourceShowDetails theDetails) throws IOException {
...
+				ourLog.warn("Received a null resource during STORAGE_PRESHOW_RESOURCES. This is a bug and should be reported. Skipping resource.");
+				continue;
+			}
+			unmarshalledByteCount = inflateBinariesInResource(unmarshalledByteCount, nextResource);
+		}
+	}
 
-			IIdType resourceId = nextResource.getIdElement();
-			List<IBinaryTarget> attachments = recursivelyScanResourceForBinaryData(nextResource);
+	private long inflateBinariesInResource(long theUnmarshalledByteCount, IBaseResource theResource) throws IOException {
```
Comment (jamesagnew) [inline hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/binary/interceptor/BinaryStorageInterceptor.java:267]: Nitpick: I think it'd be a bit easier to follow if you didn't pass theUnmarshalledByteCount in, and just returned the total for this rsources and added it to the running total outside this method

**Review by jamesagnew — APPROVED** [review]

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/binary/interceptor/BinaryStorageInterceptor.java (hunk 1 above)
Comment (tadgh) [reply hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/binary/interceptor/BinaryStorageInterceptor.java:267]: ah yeah makes sense. Laziness on my part, trusting the auto-refactor's choices. Fixing.

Comment (tadgh) [reply hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/binary/interceptor/BinaryStorageInterceptor.java:267]: Ah actually... we need to know the current total _in_ the loop for the purpose of stopping if we are over-inflating.

**Pull request merged**